opentelemetry-sdk/src/metrics/mod.rs

```diff
@@ -1032,6 +1033,61 @@ mod tests {
         assert!(resource_metrics.is_empty(), "No metrics should be exported as no new measurements were recorded since last collect.");
     }
 
+    #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+    async fn counter_multithreaded() {
```

Comment:
nit - Should we instead convert the existing single threaded test to multi-threaded. Or if we need both the tests, how about having a helper which takes number of threads, and temporality as argument?

Reply:
Maybe keep threading one separate for now, as we need to add more complexities to the threading one (introduce multiple collections etc.). Once we are confident that coverage is good, we can optimize code re-use.